api/tests/Feature/PoolCandidatesPaginatedTest.php

```diff
@@ -0,0 +1,199 @@
+    {
+        $poolOperator = User::factory()
+            ->asPoolOperator($this->team->name)
+            ->create();
+        $this->assertPaginatedResponse($poolOperator, 1, [
+            $this->teamCandidate->id,
+        ]);
+    }
+
+    public function testRequestResponserCanViewAllSubmittedApplications(): void
+    {
+        $requestResponder = User::factory()
+            ->asRequestResponder()
+            ->create();
+        $this->assertPaginatedResponse($requestResponder, 2, [
+            $this->noTeamCandidate->id,
+            $this->teamCandidate->id,
+        ]);
+    }
+
+    public function testCommunityManagerCannotViewAnyApplications(): void
+    {
+        $communityManager = User::factory()
+            ->asCommunityManager()
+            ->create();
+        $this->assertPaginatedResponse($communityManager, 0, []);
+    }
+
+    public function testProcessOperatorCanViewPoolApplications(): void
+    {
+        $processOperator = User::factory()
+            ->asProcessOperator($this->pool->id)
+            ->create();
+        $this->assertPaginatedResponse($processOperator, 1, [
+            $this->teamCandidate->id,
+        ]);
+    }
+
+    public function testUnassociatedProcessOperatorCannotViewAnyApplications(): void
+    {
+        $pool = Pool::factory()->create();
+        $processOperator = User::factory()
+            ->asProcessOperator($pool->id)
+            ->create();
+        $this->assertPaginatedResponse($processOperator, 0, []);
+    }
+
+    // TO DO: Update in #10364
+    public function testCommunityRecruiterCannotViewAnyApplications(): void
+    {
+        $community = Community::factory()->create();
+        $processOperator = User::factory()
+            ->asCommunityRecruiter($community->id)
+            ->create();
+        $this->assertPaginatedResponse($processOperator, 0, []);
+    }
+
+    // TO DO: Update in #10364
+    public function testCommunityAdminCannotViewAnyApplications(): void
+    {
+        $community = Community::factory()->create();
+        $communityAdmin = User::factory()
+            ->asCommunityAdmin($community->id)
+            ->create();
+        $this->assertPaginatedResponse($communityAdmin, 0, []);
+    }
+
+    public function testPlatformAdminCanViewAllSubmittedApplications(): void
+    {
+        $platformAdmin = User::factory()
+            ->asAdmin()
+            ->create();
+        $this->assertPaginatedResponse($platformAdmin, 2, [
+            $this->noTeamCandidate->id,
+            $this->teamCandidate->id,
+        ]);
+    }
+
+    protected function assertPaginatedResponse(User $user, int $count, array $ids): void
+    {
+        $res = $this->actingAs($user, 'api')
+            ->graphQL($this->query);
+
+        $res->assertJsonFragment([
+            'paginatorInfo' => [
+                'total' => $count,
+            ],
+        ]);
+
+        foreach ($ids as $id) {
+            $res->assertJsonFragment(['id' => $id]);
+        }
+    }
```

Comment:
I know this works, but can we move this above the tests before `setUp()` since it's being used before.

Reply:
Yup, moved up in [move assertion helper above first usage](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11022/commits/938e937881b382ae888fb141936d6909930a74b0)